### PR TITLE
fix: crash on 32bit systems during pre-allocation

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: crash on 32bit systems during pre-allocation

## Motivation and Context
pre-allocate < 2GiB on 32bit systems

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/handler-api.go [hunk 1]
```diff
@@ -87,14 +87,14 @@ func cgroupMemLimit() (limit uint64) {
...
+	available = 2048 * blockSizeV2 * 2 // Default to 4 GiB when we can't find the limits.
 
 	if runtime.GOOS == "linux" {
 		// Useful in container mode
 		limit := cgroupMemLimit()
 		if limit > 0 {
-			// A valid value is found
-			available = limit
+			// A valid value is found, return its 75%
+			available = limit * (3 / 4)
```
Comment (poornas) [inline cmd/handler-api.go]:
```suggestion
			available = uint64(float64(limit)*.75)
```

File: cmd/handler-api.go [hunk 2]
```diff
@@ -103,7 +103,8 @@ func availableMemory() (available uint64) {
 	if err != nil {
 		return
 	}
-	available = memStats.Available / 2
+	// A valid value is available return its 75%
+	available = memStats.Available * (3 / 4)
```
Comment (poornas) [inline cmd/handler-api.go]:
```suggestion
	available =  uint64(float64(memStats.Available)*.75)
```

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/handler-api.go (hunk 2 above)
Comment (klauspost) [inline cmd/handler-api.go]:
```suggestion
	available = (memStats.Available * 3) / 4
```
parenthesis for clarity

File: cmd/handler-api.go (hunk 1 above)
Comment (klauspost) [inline cmd/handler-api.go]:
```suggestion
			available = (limit * 3) / 4
```

**Review by klauspost — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**